src/commands/cloud-login.js

```diff
@@ -1,134 +1,294 @@
+      .prompt([
+        {
+          name: 'auth',
+          message: 'Select an authorization flow',
+          type: 'list',
+          choices: [
+            { name: 'Email and Password', value: 'password' },
+            { name: 'Secret', value: 'secret' },
+            // { name: 'GitHub', value: 'github' },
+            // { name: 'Netlify', value: 'netlify' },
+          ],
+        },
+      ])
+      .then(({ auth }) => {
+        this.auth = auth
+      })
+  }
+
+  async askIsDefault(endpoints) {
+    if (!this.config.default && endpoints.length === 1) {
+      await setDefaultEndpoint(endpoints[0])
+      return this.log(`Endpoint '${endpoints[0]}' added as default`)
     }
-  })
-}
 
-function secretStrategy(secret) {
-  return saveEndpointOrError(newEndpoint, alias, secret)
-}
+    if (this.config.default === endpoints[0] && endpoints.length === 1) {
+      return this.log(`Endpoint '${endpoints[0]}' added`)
+    }
+
+    // If 1 new endpoint which is not a default one (and default exists), ask a user to consider it as default
+    // If more than 1 endpoints, ask which one a user would like to be a default (or keep existing)
 
-async function emailStrategy(email) {
-  const password = await cli.prompt('Password', {
-    type: 'hide',
-    timeout: 120000,
-  })
+    const { defaultEndpoint } = await inquirer.prompt([
+      {
+        name: 'setDefault',
+        message: `Would you like endpoint '${endpoints[0]}' to be default?`,
+        type: 'confirm',
+        when: endpoints.length === 1,
+      },
+      {
+        name: 'defaultEndpoint',
+        message:
+          'Endpoints created. Would you like to set some of them as default?',
+        type: 'list',
+        when: endpoints.length > 1,
+        choices: [
+          { name: `Keep '${this.config.default}' endpoint as default` },
+          ...endpoints
+            .filter((e) => e !== this.config.default)
+            .map((e) => ({ name: e, value: e })),
+        ],
+      },
+    ])
 
-  const formData = {
-    email: email,
-    password: password,
-    session: 'Fauna Shell - ' + os.hostname(),
+    if (defaultEndpoint) {
+      return setDefaultEndpoint(defaultEndpoint)
+        .then(this.log)
+        .catch(this.error)
+    }
   }
 
-  return fetch({
-    method: 'POST',
-    uri: SHELL_LOGIN_URL,
-    form: formData,
-    resolveWithFullResponse: true,
-  })
-    .then(function (res) {
-      const secret = JSON.parse(res.body).secret
-      return saveEndpointOrError(newEndpoint, alias, secret)
+  async secretStrategy() {
+    const data = await inquirer.prompt([
+      {
+        name: 'secret',
+        message: 'Secret',
+        type: 'input',
+      },
+      {
+        name: 'region',
+        message: 'Select a region',
+        type: 'list',
+        choices: [
+          { name: 'Classic', value: 'global' },
+          { name: 'Europe (EU)', value: 'eu' },
+          { name: 'United States (US)', value: 'us' },
+        ],
+      },
+    ])
```

Comment:
Can we move the `Select a region` prompt to immediately follow the `Select an environment` prompt? Then it will make more sense when we prompt for the alias prefix.